tests/src/Context/DrupalMailContext.php

```diff
@@ -0,0 +1,163 @@
+use Drupal\DrupalDriverManager;
+use Drupal\DrupalExtension\Context\DrupalSubContextInterface;
+use Drupal\DrupalExtension\Context\RawDrupalContext;
+
+/**
+ * Context for the internal mail handling.
+ *
+ * This context can be used for testing the mail features.
+ */
+class DrupalMailContext extends RawDrupalContext implements DrupalSubContextInterface {
+  /**
+   * The Drupal driver manager.
+   *
+   * @var DrupalDriverManager
+   */
+  protected $drupal;
+
+  /**
+   * Contains original variable values set during test execution.
+   *
+   * @var array
+   */
+  protected $variables = array();
+
+  /**
+   * The captured mail data.
+   *
+   * @var array
+   */
+  protected $capturedMail;
+
+  /**
+   * The variable context.
+   *
+   * @var VariableContext
+   */
+  protected $variableContext;
+
+  /**
+   * Constructs a DrupalMailContext object.
+   *
+   * @param DrupalDriverManager $drupal
+   *   The Drupal driver manager.
+   */
+  public function __construct(DrupalDriverManager $drupal) {
+    $this->drupal = $drupal;
+  }
+
+  /**
+   * Grabs other contexts we rely on from the environment.
+   *
+   * @BeforeScenario @internalMail
+   */
+  public function gatherContexts(BeforeScenarioScope $scope) {
+    $environment = $scope->getEnvironment();
+    $this->variableContext = $environment->getContext(VariableContext::class);
+  }
+
+  /**
+   * Clear the captured data.
+   *
+   * @AfterScenario @internalMail
+   */
+  public function clearCapturedData(AfterScenarioScope $scope) {
+    // Deleting the 'drupal_test_email_collector' variable.
+    variable_del('drupal_test_email_collector');
+  }
+
+  /**
+   * Configure Drupal to use the DrupalMailContext for mail handling.
+   *
+   * @Given platform is configured to use the internal mail handling
+   */
+  public function platformIsConfiguredToUseInternalMailHandling() {
+    $mail_system = array('default-system' => 'TestingMailSystem');
+    // Using VariableContext to set the specific variable. It will be
+    // restored to the previous value after finishing a scenario.
+    $this->variableContext->setVariable('mail_system', $mail_system);
+  }
+
+  /**
+   * Verifies that Drupal internal mail handler received the mail.
+   *
+   * @Then the e-mail has been sent
+   */
+  public function assertInternalMailHandlerReceivedTheMail() {
+    $captured_mail = $this->getCapturedMail();
+
+    if (empty($captured_mail)) {
+      throw new \Exception('The Drupal internal mail handler did not receive any mail.');
+    }
+
+    $this->capturedMail = $captured_mail;
+  }
+
+  /**
+   * Assert the properties in the captured mail.
+   *
+   * @Then the sent e-mail has the following properties:
+   */
+  public function assertCapturedMailHasTheFollowingProperties(TableNode $table) {
+    $this->assertCapturedMail();
+    $expected_properties = $table->getRowsHash();
+
+    foreach ($expected_properties as $key => $value) {
+      assert($this->capturedMail[$key], equals($value));
```

Comment:
For a business use, should be better to add a message of this assert. What does you think?

Actually, the exception output is :

      Failed asserting that 'admin@example.com' is equal to <string:admin@examplee.com>.
      --- Expected
      +++ Actual
      @@ @@
      -'admin@examplee.com'
      +'admin@example.com'
       (bovigo\assert\AssertionFailure)